src/autoscaler/api/brokerserver/broker_handler.go

```diff
@@ -55,13 +58,15 @@ func (h *BrokerHandler) CreateServiceInstance(w http.ResponseWriter, r *http.Req
 	body := &models.InstanceCreationRequestBody{}
 	err := json.NewDecoder(r.Body).Decode(&body)
 	if err != nil {
+		h.logger.Error("failed to create service instance when trying to read request body", err)
 		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
 			Code:    "Interal-Server-Error",
```

Comment:
Suggest to return 400 Bad request when the request body can't be decode correctly.  The same comment will apply to other similar implementations.